### PR TITLE
datatype: Avoid NULL dereferencing during get_yaksa_type

### PR DESCRIPTION
**Pull request by sagarth**
## Pull Request Description

A simple patch to fix a Klocwork issue. 

## Author Checklist
* [x] **Provide Description** 
      Particularly focus on _why_, not _what_. Reference background, issues, test failures, xfail entries, etc.
* [x] **Commits Follow Good Practice**
      Commits are self-contained and do not do two things at once. 
      Commit message is of the form: `module: short description` 
      Commit message explains what's in the commit.
* [x] **Passes All Tests**
      Whitespace checker. Warnings test. Additional tests via comments.
* [x] **Contribution Agreement**
      For non-Argonne authors, check [contribution agreement](http://www.mpich.org/documentation/contributor-docs/). 
      If necessary, request an explicit comment from your companies PR approval manager.

### CODE REVIEW

**Review by hzhou — APPROVED** [review]
LGTM

**Comment by sagarth** [comment]
@hzhou I just rebased this PR. Do we need to run any specific test before merging?

**Comment by hzhou** [comment]
> 
> 
> @hzhou I just rebased this PR. Do we need to run any specific test before merging?

No. I'll merge it after the warnings test clear.

**Pull request merged**
